engine/src/main/java/org/terasology/i18n/TranslationProject.java

```diff
@@ -44,9 +31,10 @@
      * If no perfect match is found for the given locale, fallback strategies will attempt to find the closest match.
      * @param id the id of the string to translate (without project reference).
      * @param locale the target locale
+     * @param arguments {@link java.text.MessageFormat} arguments for the translation string
      * @return the translated string
      */
-    Optional<String> translate(Name id, Locale locale);
+    Optional<String> translate(Name id, Locale locale, Object... arguments);
```

Comment:
As you said this is definitely source compatible - I'd say time will tell if it is binary compatible 🙈